### PR TITLE
add documentation for sanitizer

### PR DESCRIPTION
**Pull request by chenfengjin**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Brief of your solution

Please describe your solution to solve the issue or feature request.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: source/development_manuals/contract-vulnerability.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    :linenos:
+
+    DEFINE_METHOD(SanitizerCase, undefined_behavior) {
+    xchain::Context* ctx = self.context();
+    int *a = 0, b;
+    b = *a;
+    ctx->ok("ok");
+    }
+
+可以看到,在代码中存在空指针解引用的，这段代码的具体行为会随着编译器，运行时甚至是变异参数的不同的不同而不同。
```
Comment (wangchao0222) [inline source/development_manuals/contract-vulnerability.rst:26]: 变异->编译
后面“的不同”重复了

File: source/development_manuals/contract-vulnerability.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+可以看到，在执行单元测试过程中，明确给出了存在缓冲区溢出的行为，并给出了对应的行号和列号。
+
+可以在 github 上找到：`完整的漏洞示例代码和测试文件 <https://github.com/xuperchain/contract-sdk-cpp/tree/main/example/sanitizer_case>`_  
+
+
+
+其他未定义行为和内存异常
+>>>>>>>>>>>>>>>>>>>>>>>>
+
+出了支持空指针接引用和缓冲区溢出，xdev 也支持其他的未定义行为和内存异常问题，如 Use After Free， Use After Return，空指针赋值等等。
```
Comment (wangchao0222) [inline source/development_manuals/contract-vulnerability.rst:186]: 出了->除了
后面建议改为：xdev也支持xxx问题的检测

File: source/development_manuals/contract-vulnerability.rst (hunk 1 above)
Comment (chenfengjin) [reply source/development_manuals/contract-vulnerability.rst]: fixed

File: source/development_manuals/contract-vulnerability.rst (hunk 2 above)
Comment (chenfengjin) [reply source/development_manuals/contract-vulnerability.rst]: fixed

**Review by wangchao0222 — APPROVED** [review]

**Pull request merged**